src/p4est_connectivity.h

```diff
@@ -446,6 +446,18 @@ sc_array_t         *p4est_connectivity_deflate (p4est_connectivity_t * conn,
                                                 p4est_connectivity_encode_t
                                                 code);
 
+/** Save a connectivity structure to disk and not close the file.
+ * \param [in] filename         Name of the file to write.
+ * \param [in] connectivity     Valid connectivity structure.
+ * \param [in,out] sink_out     A pointer to pointer to set the sc_io_sink_t.
+ * \return                      Returns 0 on success, nonzero on file error.
+ */
+int                 p4est_connectivity_save_preserve (const char *filename,
```

Comment:
Wondering: can we use p4est_connectivity_sink and then write the forest using sc_io_sink functions, to not be using sink->file? One may also ask whether we should not introduce a function p4est_sink that will share a lot of code with p4est_save.

Reply:
I can replace the `sink->file` occurrences with sc_io_sink function calls. 

However, the `save_ext` is currently using `fseek` on all ranks > 0 with respect to I/O operations that are started with `fopen` and not with `sc_io_sink_new`. To my understanding, seeking is not supported by the sc_io_sink functions.  Hence, I think that without such a functionality for sc_io_sink we can not write the forest only with sc_io_sink functions but we still can avoid `sink->file`.